tests/core/arch-arm/thumbrelocator.c

```diff
@@ -84,6 +86,89 @@ TESTCASE (handle_extended_instructions)
   g_assert_cmpint (memcmp (fixture->output, input, sizeof (input)), ==, 0);
 }
 
+TESTCASE (vldrpc_t1_should_be_rewritten)
+{
+  const guint16 input[] = {
+    GUINT16_TO_LE (0xeddf), 
+    GUINT16_TO_LE (0x0a00), /*vldr  s1, [pc, #0] */
+  };
+  const guint16 expected_output_instructions[] = {
+    GUINT16_TO_LE (0xb401), /* push {r0}          */
+    GUINT16_TO_LE (0x4802), /* ldr  r0, [pc, #8] */
+    GUINT16_TO_LE (0xedd0), /* ...                */
+    GUINT16_TO_LE (0x0a00), /* vldr s1, [r0]      */
+    GUINT16_TO_LE (0xbc01), /* pop  {r0}          */
+    GUINT16_TO_LE (0x46c0), /* nop                */
+    GUINT16_TO_LE (0xffff), /* <calculated PC     */
+    GUINT16_TO_LE (0xffff), /*  goes here>        */
+  };
+  gchar expected_output[8 * sizeof (guint16)];
+
+  guint32 calculated_pc;
+  const cs_insn * insn = NULL;
+
+  SETUP_RELOCATOR_WITH (input);
+
+  memcpy (expected_output, expected_output_instructions,
+      sizeof (expected_output_instructions));
+  calculated_pc = (fixture->rl.input_pc + 4) & ~(4 - 1);
+#if G_BYTE_ORDER == G_LITTLE_ENDIAN
```

Comment:
I suppose we don't need two codepaths here (and below), we can just assign the value without converting as we want to write the value using the native endian?